fetch_test.go

```diff
@@ -46,8 +46,8 @@ func TestFetchProtocolTrip(t *testing.T) {
 	d2 := &datastore{map[string][]byte{"key": []byte("value2")}}
 	h2 := newFetchProtocol(ctx, hosts[1], d2.Lookup)
 
-	fetchCheck(t, ctx, h1, h2, "key", []byte("value2"))
-	fetchCheck(t, ctx, h2, h1, "key", []byte("value1"))
+	fetchCheck(ctx, t, h1, h2, "key", []byte("value2"))
```

Comment:
Could move this refactor to a separate PR if we wanted, but it's pretty simple.

Reply:
Np. Small cleanups are almost always fine to just include.